applicationLibrary/molecularDynamics/molecularDynamicsLibrary/AxilrodTellerMutoFunctorHWY.h

```diff
@@ -0,0 +1,1836 @@
+    for (size_t i = soaSize - 1; i > 1; --i) {
+      const auto ownedStateI = ownedStatePtr[i];
+      if (ownedStateI == autopas::OwnershipState::dummy) {
+        continue;
+      }
+      VectorDouble fXAccI = _zeroDoubleVec;
+      VectorDouble fYAccI = _zeroDoubleVec;
+      VectorDouble fZAccI = _zeroDoubleVec;
+
+      for (size_t j = i - 1; j > 0; --j) {
+        const auto ownedStateJ = ownedStatePtr[j];
+        if (ownedStateJ == autopas::OwnershipState::dummy) {
+          continue;
+        }
+
+        // Load precomputed distances for particle i and j.
+        auto [distXIJ, distYIJ, distZIJ, distSquaredIJ, invR5IJ] = intraSoADists.get(i, j);
+
+        if (distSquaredIJ > cutoffSquared) {
+          continue;
+        }
+
+        // prepare vector registers
+        VectorDouble fXAccJ = _zeroDoubleVec;
+        VectorDouble fYAccJ = _zeroDoubleVec;
+        VectorDouble fZAccJ = _zeroDoubleVec;
+
+        const auto distXIJVec = highway::Set(tag_double, distXIJ);
+        const auto distYIJVec = highway::Set(tag_double, distYIJ);
+        const auto distZIJVec = highway::Set(tag_double, distZIJ);
+        const auto distSquaredIJVec = highway::Set(tag_double, distSquaredIJ);
+        const auto invR5IJVec = highway::Set(tag_double, invR5IJ);
+
+        const size_t blockEnd = j & ~(_vecLengthDouble - 1);
+
+        unsigned int k = 0;
+        for (; k < blockEnd; k += _vecLengthDouble) {
+          handleKLoopBody<DistanceMatrix<true, alignedSoAView>, DistanceMatrix<true, alignedSoAView>, /*newton3*/ true,
+                          /*newton3Kernel*/ true,
+                          /*remainder*/ false, alignedSoAView>(
+              i, j, k, const_nu, intraSoADists, intraSoADists, ownedStatePtr, typePtr, typePtr, typePtr, distXIJVec,
+              distYIJVec, distZIJVec, distSquaredIJVec, invR5IJVec, fXAccI, fYAccI, fZAccI, fXAccJ, fYAccJ, fZAccJ,
+              ownedStateI, ownedStateJ, fxPtr, fyPtr, fzPtr, virialSumX, virialSumY, virialSumZ, potentialEnergySum,
+              numKernelCallsN3Sum, numGlobalCalcsN3Sum, numKernelCallsNoN3Sum, numGlobalCalcsNoN3Sum);
+        }
+
+        const auto restK = j - k;
+
+        if (restK > 0) {
+          handleKLoopBody<DistanceMatrix<true, alignedSoAView>, DistanceMatrix<true, alignedSoAView>, /*newton3*/ true,
+                          /*newton3Kernel*/ true,
+                          /*remainder*/ true, alignedSoAView>(
+              i, j, k, const_nu, intraSoADists, intraSoADists, ownedStatePtr, typePtr, typePtr, typePtr, distXIJVec,
+              distYIJVec, distZIJVec, distSquaredIJVec, invR5IJVec, fXAccI, fYAccI, fZAccI, fXAccJ, fYAccJ, fZAccJ,
+              ownedStateI, ownedStateJ, fxPtr, fyPtr, fzPtr, virialSumX, virialSumY, virialSumZ, potentialEnergySum,
+              numKernelCallsN3Sum, numGlobalCalcsN3Sum, numKernelCallsNoN3Sum, numGlobalCalcsNoN3Sum, restK);
+        }
```

Comment:
In theory, with proper aligned vectors, we don't need this distinction, which makes our code simpler. The vectors are padded with zeros (or just something arbitrary) at the end, so the SoA length is perfectly divisible by the vector register size. The only difference in handling should be that these extra elements are dummies or otherwise have no impact, and that they are not ultimately converted back to AoS elements.